xmtp_api_grpc/src/grpc_api_helper.rs

```diff
@@ -32,8 +32,11 @@ use xmtp_proto::{
 async fn create_tls_channel(address: String) -> Result<Channel, Error> {
     let channel = Channel::from_shared(address)
         .map_err(|e| Error::new(ErrorKind::SetupError).with(e))?
+        .initial_connection_window_size(Some((1 << 31) - 1))
         .keep_alive_while_idle(true)
-        .connect_timeout(Duration::from_secs(5))
+        .connect_timeout(Duration::from_secs(10))
+        .tcp_keepalive(Some(Duration::from_secs(15)))
```

Comment:
**Purpose:** Configures the TCP keep-alive interval for the socket connection.

**Functionality:** This setting tells the operating system to send TCP keep-alive probes periodically when no data has been transferred over the connection within the specified interval.

**Impact:** Similar to the gRPC-level keep-alive, this helps keep the connection alive at the TCP layer and detect broken connections. It's particularly useful for detecting half-open connections and ensuring that resources are not wasted on unresponsive peers.